### ISSUE TITLE
Error when executing DataStore.start() and DataStore.clear()

### ISSUE THREAD

**Issue by fitzglitz** [aws-amplify/amplify-js#10641]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React Native

### Amplify APIs

Authentication, DataStore

### Amplify Categories

auth, storage

### Environment information

<details>

```
# Put output below this line
  System:
    OS: macOS 12.6.1
    CPU: (8) x64 Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
    Memory: 76.00 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 16.15.0 - /usr/local/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 8.19.2 - /usr/local/bin/npm
    Watchman: 2022.10.31.00 - /usr/local/bin/watchman
  Browsers:
    Chrome: 107.0.5304.87
    Firefox: 96.0.1
    Safari: 16.1
    Safari Technology Preview: 16.4
  npmPackages:
    @azure/core-asynciterator-polyfill: ^1.0.2 => 1.0.2
    @babel/core: ^7.12.9 => 7.20.2
    @babel/runtime: ^7.12.5 => 7.20.1
    @gorhom/bottom-sheet: ^4 => 4.4.5
    @react-native-async-storage/async-storage: ^1.17.10 => 1.17.10
    @react-native-community/eslint-config: ^2.0.0 => 2.0.0
    @react-native-community/geolocation: ^3.0.1 => 3.0.3
    @react-native-community/netinfo: ^9.3.5 => 9.3.6
    @react-native-picker/picker: ^2.4.8 => 2.4.8
    @react-navigation/bottom-tabs: ^6.4.0 => 6.4.0
    @react-navigation/native: ^6.0.13 => 6.0.13
    @react-navigation/native-stack: ^6.9.0 => 6.9.1
    HelloWorld:  0.0.1
    amazon-cognito-identity-js: ^5.2.12 => 5.2.12
    aws-amplify: ^4.3.43 => 4.3.43
    aws-amplify-react-native: ^6.0.8 => 6.0.8
    babel-jest: ^26.6.3 => 26.6.3
    eslint: ^7.32.0 => 7.32.0
    hermes-inspector-msggen:  1.0.0
    jest: ^26.6.3 => 26.6.3
    metro-react-native-babel-preset: ^0.72.1 => 0.72.3
    react: ^18.1.0 => 18.2.0
    react-dom: ^17.0.2 => 17.0.2
    react-native: ^0.70.4 => 0.70.5
    react-native-geolocation-service: ^5.3.1 => 5.3.1
    react-native-gesture-handler: ^2.7.1 => 2.8.0
    react-native-google-places-autocomplete: ^2.4.1 => 2.5.1
    react-native-maps: ^1.3.2 => 1.3.2
    react-native-ratings: ^8.1.0 => 8.1.0
    react-native-reanimated: ^2.11.0 => 2.12.0
    react-native-safe-area-context: ^4.3.4 => 4.4.1
    react-native-screens: ^3.17.0 => 3.18.2
    react-native-vector-icons: ^9.2.0 => 9.2.0
    react-test-renderer: 18.1.0 => 18.1.0
  npmGlobalPackages:
    @aws-amplify/cli: 10.4.0
    bower: 1.8.13
    cordova: 9.0.0
    corepack: 0.10.0
    forcedroid: 6.2.0
    forcehybrid: 8.1.0
    generator-office: 1.9.1
    ios-deploy: 1.9.2
    npm: 8.19.2
    tar: 6.1.11
    yarn: 1.22.17
    yo: 4.3.0

```

</details>

### Describe the bug

In my react native app, I need to use both authentication and datastore.  Because it's react native, I can not use the withAuthenticator included in "aws-amplify" as it is not available yet, which forces me to use withAuthenticator from "aws-amplify-react-native".

The problem is that there is a DataStore included in both "aws-amplify" and "aws-amplify-react-native". I'm only importing and using DataStore from "aws-amplify" in the app.

Whenever I try to execute DataStore.start() or DataStore.clear(), I get the following error:
DataStore - Schema is not initialized. DataStore will not function as expected. This could happen if you have multiple versions of DataStore installed. Please see https://docs.amplify.aws/lib/troubleshooting/upgrading/q/platform/js/#check-for-duplicate-versions
I've executed the npm command listed on the link above, which is used to identify if there are multiple version of DataStore installed.  The command returns **aws-amplify-react-native**, which tells me there are duplicates.

So, I need "aws-amplify-react-native" for with Authenticator, and I need "aws-amplify" for the latest DataStore functionality, but the 2 libraries do not seem to coexist very well since DataStore exists in both.

I do not want to go back the using DataStore from "aws-amplify-react-native" as support is already slated for deprecation.

What can be done?

### Expected behavior

Since with Authenticator is not available for react native in "aws-amplify", I need to be able use the withAuthenticator from "aws-amplify-react-native" and also be able to DataStore from "aws-amplify".

### Reproduction steps

No specific steps.

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by fitzglitz** [comment]
One other interesting note about this issue.

If I do not execute DataStore.start() or DataStore.clear(), and I just start executing queries to select or add records, they data is persisted correctly. But because users need to log out from the app, and a different user can log in, I really need to clear the datastore.

**Comment by fitzglitz** [comment]
Chris, thanks for your assistance.
Here is how I've configured Amplify and DataStore, and calling the configure methods.

`
import {Amplify, DataStore} from 'aws-amplify';
import {withAuthenticator, AmplifyTheme} from 'aws-amplify-react-native';
import awsExports from '../../src/aws-exports';

//Amplify.Logger.LOG_LEVEL = 'DEBUG';
Amplify.configure({
  ...awsExports,
  Analytics: {
    disabled: true,
  },
});
DataStore.configure();
`

I call this method to initialize DataStore
`
const initDataStore = async () => {
  console.log('initDataStore');
  try {
    await DataStore.start();
  } catch (error) {
    console.error(error);
  }
};
`

This is the error from DataStore.start()
DataStore - Schema is not initialized. DataStore will not function as expected. This could happen if you have multiple versions of DataStore installed. Please see https://docs.amplify.aws/lib/troubleshooting/upgrading/q/platform/js/#check-for-duplicate-versions

According to the link, when I run the npm command from the link and I have output, that tells me there is a DatatStore duplicate issue, and the output is "aws-amplify-react-native", that tells me the duplicate issue is with the aws-amplify-react-native library.

Let me know if you need any more info.

**Comment by chrisbonifacio** [comment]
@fitzglitz thanks for the quick reply!

Curious - why are you calling DataStore.configure separately from Amplify.configure and without any arguments?

You don't have to call `DataStore.configure` if you've already called `Amplify.configure`. If you still need to configure some DataStore settings, you can do so nested within the `Amplify.configure` similar to how you've configured Analytics.

Lastly, Make sure you are at least importing a model so that `initSchema` is called. If you look at the `models/index.js` file, you'll see that it is called before exporting the models.

**Comment by fitzglitz** [comment]
Good point. I was looking at DataStore.configure() initially to use the SQLiteAdapter. I opted to not over complicate the initial configuration of DataStore. I've now removed DataStore.configure() from the code.

I am importing a model but didn't include it in the provided code because it was just an unused import - now I know.  Here is a better representation of the current code.

`
import {Amplify, DataStore} from 'aws-amplify';
import {withAuthenticator, AmplifyTheme} from 'aws-amplify-react-native';
import awsExports from '../../src/aws-exports';
import {UserStores} from '../../src/models';

Amplify.configure({
  ...awsExports,
  Analytics: {
    disabled: true,
  },
});

const initDataStore = async () => {
  console.log('initDataStore');
  try {
    await DataStore.start();
  } catch (error) {
    console.error(error);
  }
};
`

Still get the error.  Is there anything else I need to do with the schema other than have it as an unused import?

**Comment by chrisbonifacio** [comment]
> Still get the error. Is there anything else I need to do with the schema other than have it as an unused import?

You probably just need to use an imported model at least in some capacity.

For example:

unused imported model

```js
import { Post } from "./models" // unused
```
<img width="1452" alt="Screenshot 2022-11-15 at 11 43 20 AM" src="https://user-images.githubusercontent.com/16952986/201977694-58ba8111-465b-432a-bf4e-bc29b75318f2.png">

using imported model
```js
import { Post } from "./models"

...

  const getPosts = async () => {
    try {
      const posts = await DataStore.query(Post);
      console.log(posts);
    } catch (error) {
      console.error(error);
    }
  };
  ```
  
  
<img width="1452" alt="Screenshot 2022-11-15 at 11 45 07 AM" src="https://user-images.githubusercontent.com/16952986/201978031-d15c90a8-fb79-495d-b349-a9838efc13a4.png">

**Comment by fitzglitz** [comment]
Yes, if I execute a query against the imported model, the datastore starts on its own and I am able to fully use datastore.  It's just really weird that datastore.start() fails.

I guess I'll just move past this with executing a query.

Thanks for you help!

**Comment by chrisbonifacio** [comment]
No problem! Just to be clear, the query doesn't need to be executed for DataStore.start() to work. I just added that show that I was using the import by referencing it in a function. I was still calling DataStore.start() in a useEffect and whether I got the "Schema was not initialized" error depended on whether an imported model was being lazy loaded or not.

That being said, it is okay to implicitly start DataStore by performing a query. The sync process is started by any DataStore method, DataStore.start() just requires that initSchema was called first.
